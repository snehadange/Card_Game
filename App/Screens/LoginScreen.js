import React from 'react'
import {
    StyleSheet, View, Text, ImageBackground, Image, TextInput, AsyncStorage, Alert,
    TouchableOpacity
} from 'react-native';
import { IconsPath } from '../Assets';
import { Ionicons, MaterialIcons } from 'react-native-vector-icons';


export default class LoginScreen extends React.Component {
    constructor(props) {
        super(props);

        this.state = {
            icEye: 'visibility-off',
            showPassword: true,
            email: '',
            password: '',
            emailError: '',
            passwordError: ''
        }

        this.baseurl = "http://192.168.1.102:8000/src/router"
     

    }



    validateUserLogin = (type) => {
        let emailrjx = /^\w+([\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w{2,3})+$/;
        let passwordrjx = /^(?=.*[A-Za-z])(?=.*\d)[A-Za-z\d]{8,}$/



        if (type == 'email') {
            if (emailrjx.test(this.state.email)) {
                this.setState({
                    emailError: ''
                })
            } else {
                this.setState({
                    emailError: 'Please enter valid email address'
                })
            }
        }
        else if (type == 'password') {
            if (passwordrjx.test(this.state.password)) {
                this.setState({
                    passwordError: ''
                })
            } else {
                this.setState({
                    passwordError: 'Password must be Minimum eight characters, at least one letter and one number '
                })
            }
        }


    }

    storeData = () => {
        const { navigate } = this.props.navigation;


        let login_collection = {
            email: this.state.email,
            password: this.state.password
        }

        let url = this.baseurl + '/login'
        console.log(url)
        if (this.state.email == '') {
            alert("Please enter valid email")
        } else if (this.state.password == '') {
            alert("Please enter valid password")
        } else {

            fetch(url,
                {
                    method: 'POST', // or 'PUT'
                    headers: {
                        'Accept': 'application/json',
                        'Content-Type': 'application/json'
                    },
                    body: JSON.stringify(login_collection),

                })

                .then(response => response.json())
                .then((data) => {


                    if (data.message == 'Invalid email/password!!!') {
                        console.log("Invalid Login:", data)
                        Alert.alert(
                            'Oops!',
                            'Invalid Email/password'
                        );
                    }
                    console.log("Login suceessfull :", data)
                    this.setState({
                        email: '',
                        password: '',
                    })

                    AsyncStorage.setItem('tokenKey', data.data.token)
                    AsyncStorage.setItem('username', data.data.user.name)



                    if (data.erros) {


                        data.errors.map((err) => {

                            if (err.param == 'email') {
                                this.setState({
                                    emailError: err.msg
                                })
                            } else if (err.param == 'password') {
                                this.setState({
                                    passwordError: err.msg
                                })
                            }
                        })
                    }
                    else {
                        Alert.alert(
                            'Great!',
                            ' Login Successful',
                            [
                                { text: 'OK', onPress: () => navigate('ProfileScreen') },
                            ]
                        );
                    }




                }).catch((error) => console.log("Erros:", error))
        }
    }


    changePwdType = () => {

        if (this.state.showPassword) {
            this.setState({
                icEye: 'visibility',
                showPassword: false,
                password: this.state.password
            })
        } else {
            this.setState({
                icEye: 'visibility-off',
                showPassword: true,
                password: this.state.password
            })
        }

    };



    render() {

        return (

            <ImageBackground source={IconsPath.back} style={styles.container}>


                <Ionicons name="ios-mail" size={30} style={{ marginLeft: -190, top: 80 }} />
                <Ionicons name="ios-lock" size={30} style={{ marginLeft: -190, top: 120 }} />

                <TextInput style={styles.input}
                    underlineColorAndroid="transparent"
                    placeholder="email"
                    placeholderTextColor="black"
                    autoCapitalize='none'
                    value={this.state.email}
                    disableFullscreenUI={true}
                    onEndEditing={() => this.validateUserLogin('email')}
                    onChangeText={(text) => this.setState({ email: text })}
                />
                <Text style={{ top: 10, color: 'red', fontWeight: 'bold' }}>{this.state.emailError}</Text>

                <TextInput style={styles.input}
                    underlineColorAndroid="transparent"
                    placeholder="password"
                    placeholderTextColor="black"
                    secureTextEntry={this.state.showPassword}
                    autoCapitalize='none'
                    value={this.state.password}
                    disableFullscreenUI={true}
                    onEndEditing={() => this.validateUserLogin('password')}
                    onChangeText={(text) => this.setState({ password: text })}
                />
                <MaterialIcons
                    style={styles.icon}
                    name={this.state.icEye}
                    size={30}
                    onPress={this.changePwdType}
                />
                <Text style={{ top: 10, color: 'red', fontWeight: 'bold' }}>{this.state.passwordError}</Text>



                <View style={{ top: 10, backgroundColor: '#3399FF', width: 250, height: 50, borderRadius: 25 }}>
                    <TouchableOpacity style={{ top: 10, left: 90 }} onPress={() => this.storeData()}>
                        <Text style={{ color: 'white', fontWeight: 'bold', fontSize: 20 }}>Login</Text>
                    </TouchableOpacity>
                </View>

                <View style={{ top: 30, flexDirection: 'row' }} >
                    <Text style={{ color: 'white', fontWeight: 'bold' }}>Don't have an account?</Text>
                    <TouchableOpacity onPress={() => this.props.navigation.navigate('Registration')}>
                        <Text style={{ color: 'blue', fontWeight: 'bold' }}>Signup</Text>
                    </TouchableOpacity>
                </View>
            </ImageBackground>



        )
    }
}

const styles = StyleSheet.create({
    container: {
        flex: 4,
        flexDirection: 'column',
        alignItems: 'center',
        justifyContent: 'center',
        width: '100%', height: '100%',

    },
    input: {
        //  margin: 15,
        width: 250,
        height: 50,
        fontSize: 16,
        paddingLeft: 60,
        borderWidth: 1,
        color: 'black',
        backgroundColor: 'white',
        fontWeight: 'bold',
        top: 10,
        opacity: 0.4,
        borderRadius: 25,
    },
    error: {
        borderColor: 'red',
        borderWidth: 3
    },
    icon: {
        top: -30,
        left: 90,

    }
})


