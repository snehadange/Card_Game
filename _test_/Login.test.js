import React from 'react';
import renderer from 'react-test-renderer';
import fetch from 'node-fetch'
import fetchMock from 'fetch-mock';
import LoginScreen from '../App/Screens/LoginScreen'




const data = {
  "email": 'react@gmail.com',
  "password":'react1234'
}

it('works', () => {
  expect(1).toBe(1);
});

//snapshot testing for react Component
test("Login screen",()=>{
  const snap=renderer.create(
     <LoginScreen/>
  ).toJSON();
  expect(snap).toMatchSnapshot()
}) 

describe("isAuthenticated", () => {
  // resets your mocks so tests don't bleed into each other

  beforeEach(() => {
    fetchMock.reset();
  })
  afterEach(() => {
    fetchMock.restore();

  });

  test("success",async () => {
    fetchMock.once("http://localhost:8000/src/router/login",{
      status:200,
      body: JSON.stringify("Login:"),
      statusText: 'success',  
      headers: {'Content-Type': 'application/json'},
    });
    const res = await fetch("http://localhost:8000/src/router/login",{
      headers:{
        'Content-Type': 'application/json',
         },
       method:'post',
       body: JSON.stringify(data),
    }).then(function (res) {
      expect(res.status).toEqual(200); // Pass
      return res.json(); // return here
    }).then((json)=>{
      console.log(json); // print undefine
      expect(json).toEqual(json); 
    })
    
  })

})











