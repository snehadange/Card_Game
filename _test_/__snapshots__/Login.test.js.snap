// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login screen 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    Object {
      "alignItems": "center",
      "flex": 4,
      "flexDirection": "column",
      "height": "100%",
      "justifyContent": "center",
      "width": "100%",
    }
  }
>
  <Image
    source={1}
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": "100%",
          "width": "100%",
        },
        undefined,
      ]
    }
  />
  <Text />
  <Text />
  <TextInput
    allowFontScaling={true}
    autoCapitalize="none"
    disableFullscreenUI={true}
    onChangeText={[Function]}
    onEndEditing={[Function]}
    placeholder="email"
    placeholderTextColor="black"
    rejectResponderTermination={true}
    style={
      Object {
        "backgroundColor": "white",
        "borderRadius": 25,
        "borderWidth": 1,
        "color": "black",
        "fontSize": 16,
        "fontWeight": "bold",
        "height": 50,
        "opacity": 0.4,
        "paddingLeft": 60,
        "top": 10,
        "width": 250,
      }
    }
    underlineColorAndroid="transparent"
    value=""
  />
  <Text
    style={
      Object {
        "color": "red",
        "fontWeight": "bold",
        "top": 10,
      }
    }
  >
    
  </Text>
  <TextInput
    allowFontScaling={true}
    autoCapitalize="none"
    disableFullscreenUI={true}
    onChangeText={[Function]}
    onEndEditing={[Function]}
    placeholder="password"
    placeholderTextColor="black"
    rejectResponderTermination={true}
    style={
      Object {
        "backgroundColor": "white",
        "borderRadius": 25,
        "borderWidth": 1,
        "color": "black",
        "fontSize": 16,
        "fontWeight": "bold",
        "height": 50,
        "opacity": 0.4,
        "paddingLeft": 60,
        "top": 10,
        "width": 250,
      }
    }
    underlineColorAndroid="transparent"
    value=""
  />
  <Text
    style={
      Object {
        "color": "red",
        "fontWeight": "bold",
        "top": 10,
      }
    }
  >
    
  </Text>
  <View
    style={
      Object {
        "backgroundColor": "#3399FF",
        "borderRadius": 25,
        "height": 50,
        "top": 10,
        "width": 250,
      }
    }
  >
    <RNGestureHandlerButton
      collapsable={false}
      onGestureEvent={[Function]}
      onGestureHandlerEvent={[Function]}
      onGestureHandlerStateChange={[Function]}
      onHandlerStateChange={[Function]}
      rippleColor={0}
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          undefined,
        ]
      }
    >
      <View
        accessible={true}
        style={
          Object {
            "left": 90,
            "opacity": 1,
            "top": 10,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 20,
              "fontWeight": "bold",
            }
          }
        >
          Login
        </Text>
      </View>
    </RNGestureHandlerButton>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "top": 30,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontWeight": "bold",
        }
      }
    >
      Don't have an account?
    </Text>
    <RNGestureHandlerButton
      collapsable={false}
      onGestureEvent={[Function]}
      onGestureHandlerEvent={[Function]}
      onGestureHandlerStateChange={[Function]}
      onHandlerStateChange={[Function]}
      rippleColor={0}
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          undefined,
        ]
      }
    >
      <View
        accessible={true}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "blue",
              "fontWeight": "bold",
            }
          }
        >
          Signup
        </Text>
      </View>
    </RNGestureHandlerButton>
  </View>
</View>
`;
